ambry-store/src/main/java/com.github.ambry.store/PersistentIndex.java

```diff
@@ -80,6 +80,7 @@ public boolean accept(File dir, String name) {
   final ConcurrentSkipListMap<Offset, IndexSegment> indexes = new ConcurrentSkipListMap<>();
   final Journal journal;
   final HardDeleter hardDeleter;
+  volatile Thread hardDeleteThread = null;
```

Comment:
Since it was accessed outside of PersistentIndex, thought volatile might be required. Will remove it. But can't be made final since the thread itself will be created only if hard delete is enabled.